doc/adrs/202_add_support_for_project_duplication.adoc

```diff
@@ -0,0 +1,63 @@
+= ADR-202 - Add support for project duplication
+
+== Context
+
+An contextual menu action in the `projects` page is provided to duplicate an existing project.
+
+== Decision
+
+We will add an API to duplicate a project.
+
+=== GraphQL shema
+
+We will add a new mutation in `sirius-web.graphqls` and the corresponding input/payload.
+
+----
+extend type Mutation {
+  // [...]
+  duplicateProject(input: DuplicateProjectInput!): DuplicateProjectPayload!
+}
+
+input DuplicateProjectInput {
+  id: ID!
+  projectId: ID!
+}
+
+union DuplicateProjectPayload = ErrorPayload | DuplicateProjectSuccessPayload
+
+type DuplicateProjectSuccessPayload {
+  id: ID!
+  project: Project!
+}
+----
+
+
+=== Backend
+
+We will create a new service in Sirius Web to duplicate a project.
+This service will take care of duplicating the original project's:
+
+- Semantic Data (including domains)
+- Representations
+- Nature
+- Settings
+
+The created project should have a different ID than the original project.
+The elements contained in the created project's semantic data should have different IDs than the ones in the original project.
+The target objects of the representations should be adapted to target the duplicated objects (with their new ID).
+
+Avoiding ID duplication (for project and elements) is key to avoid clashes in library contexts.
```

Comment:
To me this is important to avoid issues if at some point the user decides to publish the duplicated project as a library. 
If they then import it in the original project we'll have clashes between the IDs of the project's elements and the library's elements.
This is maybe a corner case though, if so we can maybe get rid of it

Reply:
For me it is critical, the behavior should thus be the same as download + upload of the project. It may even require a refactoring of the download / upload code to literally use the same code between both features "duplicate" and "download/upload".

There have been multiple set of improvements on this download/upload code especially with regards to the generation of new identifier and the stability of the link between representations and the semantic data very recently. You don't want to face those problems once again.

Reply:
Done, I mentioned to reuse the existing "upload" feature, and explicitly stated that a refactoring of the existing code may be necessary to extract common behavior.